main.go

```diff
@@ -74,12 +74,82 @@ func (s *serialPort) Write(b []byte) (int, error) {
 	return p.Write(b)
 }
 
+var deferredFuncs []func()
+
 func run(ctx context.Context) {
+	// Remove any existing restart file.
+	if err := os.Remove(config.RestartFile()); err != nil && !os.IsNotExist(err) {
```

Comment:
Why do we need to do this and not the OS itself?

Reply:
Not sure what you mean, the restart file is a file only known to the agent, getting each OS service manager to do this would be annoying and probably not even possible on some systems